third_party/2and3/click_spinner/__init__.pyi

```diff
@@ -0,0 +1,34 @@
+import threading
+from types import TracebackType
+from typing import IO, Iterator, Optional, Type
+from typing_extensions import Literal
+
+__version__: str
+
+class Spinner:
+    spinner_cycle: Iterator[str] = ...
+    disable: bool = ...
+    beep: bool = ...
+    force: bool = ...
+    stream: IO[str] = ...
```

Comment:
While it is okayish to use `IO[str]`, it would be better to use a custom protocol with the `write(str)`, `flush()`, and `isatty()` methods.

Reply:
Thank you for the review @srittau.

I wrote a basic `_Stream` class based on what the docs say for [simple user-defined protocols](https://mypy.readthedocs.io/en/stable/protocols.html#simple-user-defined-protocols), but you know a lot more about this than I do, so let me know if I need to correct anything and I'll be happy to do so.

Thanks for your help.

Reply:
Looks good, thanks!